src/components/announcementsList/index.tsx

```diff
@@ -68,23 +69,23 @@ export interface AnnouncementsListProps {
 const AnnouncementsList: React.FC<AnnouncementsListProps> = ({
   announcements,
 }) => {
-  const [announcementsPerPage, setAnnouncementsPerPage] = useState<number>(0);
+  const announcementsPerPage = 6;
   const [currentPage, setCurrentPage] = useState<number>(1);
 
-  const handleNextPage = (event: any) => {
+  const handleNextPage = () => {
     setCurrentPage(currentPage + 1);
```

Comment:
To avoid race conditions, maybe consider passing in a function to the setter. Something like:
`setCurrentPage((prevState) => prevState + 1);`

This could apply to all of the handlers in this file.

Reply:
Great comment @anirudh723